### ISSUE TITLE
Line continuations aren't fully supported for per-requirement options in requirements.txt

### ISSUE THREAD

**Issue by benji-york** [tweag/FawltyDeps#225]
**Describe the bug**
Parsing of requirements.txt appears to not handle line continuations.

**To Reproduce**
Create a `requirements.txt` file with these contents:

```
black==23.1.0 \
    --hash=sha256:0052dba51dec07ed029ed61b18183942043e00008ec65d5028814afaab9a22fd
```

Run fawltydeps with this switch: `--check-unused`

Observe the output:
```
WARNING:fawltydeps.extract_declared_dependencies:Could not parse requirements.txt line 'black==23.1.0 \\': not enough values to unpack (expected 1, got 0)
```

**Expected behavior**
Correct parsing of above `requirements.txt`.

**Environment**
 - OS name + version: macOS 13.1
 - Version of the code: 0.6.1

**Additional context**
None

**Comment by jherland** [comment]
Thanks for reporting!

We have some work to do to improve our parsing of `requirements.txt` files. We'll look into it as time permits.

**Comment by Nour-Mws** [comment]
This was previously reported in #114 and partially resolved in #182.

**Comment by benji-york** [comment]
🥇

**Comment by vreuter** [comment]
@benji-york the previous version of the code which addressed per-requirement options / multi-line declarations of dependencies did not account for the possibility that there would be none of the options on the first line. Sorry that this may've caused a hiccup. Thank you for reporting this so that we could fix it.

**Comment by jherland** [comment]
I just released v0.7.0 to PyPI, where this is fixed.
